x/pki/x509/x509.go

```diff
@@ -52,9 +53,53 @@ func DecodeX509Certificate(pemCertificate string) (*X509Certificate, error) {
 		Certificate:    cert,
 	}
 
+	certificate = PatchCertificate(certificate)
+
 	return &certificate, nil
 }
 
+// This function is needed to patch the Issuer/Subject(vid/pid) field of certificate to hex format.
+// https://github.com/zigbee-alliance/distributed-compliance-ledger/issues/270
+func PatchCertificate(certificate X509Certificate) X509Certificate {
+	oldVIDKey := "1.3.6.1.4.1.37244.2.1"
+	oldPIDKey := "1.3.6.1.4.1.37244.2.2"
+
+	newVIDKey := "vid"
+	newPIDKey := "pid"
+
+	issuer := certificate.Issuer
+	issuer = FormatOID(issuer, oldVIDKey, newVIDKey)
+	issuer = FormatOID(issuer, oldPIDKey, newPIDKey)
+
+	subject := certificate.Subject
+	subject = FormatOID(subject, oldVIDKey, newVIDKey)
+	subject = FormatOID(subject, oldPIDKey, newPIDKey)
+
+	certificate.Issuer = issuer
+	certificate.Subject = subject
+
+	return certificate
+}
+
+func FormatOID(header, oldKey, newKey string) string {
+	subjectValues := strings.Split(header, ",")
+
+	for index, value := range subjectValues {
+		if strings.HasPrefix(value, oldKey) {
+			// get value from header
+			value = value[len(value)-8:]
+
+			decoded, _ := hex.DecodeString(value)
+			hexStr := "=0x" + string(decoded)
+
+			value = newKey + hexStr
```

Comment:
I think it would be more clear to use `fmt.Sprintf`:
`fmt.Sprintf("%s=0x%s", newKey, hex.DecodeString(value))`

Reply:
Done! ✅